### PR TITLE
Angulartics

### PR DESCRIPTION
**Pull request by bex1**
Introduces use of angulartics to provide support for GTM events.
Several events for common actions are now reported.
See AnalyticsActions in analytics.service.ts for reference on what the event action are.

### CODE REVIEW

File: src/app/services/analytics.service.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    SignForJobTry = "SignForJobTry",
+    UpdateSubscriptionFail = "UpdateSubscriptionFail",
+    UpdateSubscriptionSuccess = "UpdateSubscriptionSuccess",
+    UpdateSubscriptionTry = "UpdateSubscriptionTry",
+    UpdateUserFail = "UpdateUserFail",
+    UpdateUserSuccess = "UpdateUserSuccess",
+    UpdateUserTry = "UpdateUserTry",
+    UploadResumeFail = "UploadResumeFail",
+    UploadResumeSuccess = "UploadResumeSuccess",
+    UploadResumeTry = "UploadResumeTry",
```
Comment (buren) [inline src/app/services/analytics.service.ts:51]: Maybe add some type information? Just for some nice code-completion etc :)

File: src/app/components/links/basic-link/basic-link.component.ts [hunk 2]
```diff
@@ -9,6 +9,12 @@ import {UserResolver} from '../../../resolvers/user/user.resolver';
   styleUrls: ['./basic-link.component.scss'],
   template: `
     <a
+      [angulartics2On]="angulartics2On"
+      [angularticsAction]="angularticsAction"
+      [angularticsCategory]="angularticsCategory"
+      [angularticsLabel]="angularticsLabel"
+      [angularticsProperties]="angularticsProperties"
+      [angularticsValue]="angularticsValue"
```
Comment (buren) [inline src/app/components/links/basic-link/basic-link.component.ts]: Can we use a directive here? I have no idea what I'm talking about but bear with me.. 
But as I recall directives in new angular versions we could use that to create something like `just-analytics`
which could support all the above ☝️ attributes. See ng-docs @ https://angular.io/guide/attribute-directives. I'm not at all sure if this is how it works, or if its even worth it :) Just a thought and I'm curious about what you think @bex1 :) feel free to ignore this..

**Review by buren — COMMENTED** [review]
LGTM 👍 

Just to verify `this.analyticsService.publishEvent` is async right? 😉

File: src/app/services/analytics.service.ts (hunk 1 above)
Comment (bex1) [reply src/app/services/analytics.service.ts:51]: Its a Enum of strings so the type is AnalyticsAction.

**Review by buren — COMMENTED** [review]
:)

**Pull request merged**